src/conditions/check_fields.rs

```diff
@@ -432,6 +432,7 @@ impl CheckFieldsPredicate for LengthEqualsPredicate {
             Event::Log(l) => l.get(&self.target).map_or(false, |v| {
                 let len = match v {
                     Value::Array(value) => value.len(),
+                    Value::Map(value) => value.len(),
                     value => value.to_string_lossy().len(),
                 };
```

Comment:
Should this have a special case for `Value::Bytes`, as it can skip the `.into_owned()` step for valid UTF-8 strings (ie the common case). Also, this will end up producing a length of 6 for `null` which seems nonsensical. (Yes, I know this isn't directly related to this PR)

Reply:
This is only used for testing, right? I'm not too concerned about the extra `into_owned`. I can add a special case for `Null` to return `0` 👍